framework/src/modules/auth/module.ts

```diff
@@ -33,10 +43,144 @@ export class AuthModule extends BaseModule {
 	public configSchema = configSchema;
 	public commands = [new RegisterMultisignatureCommand(this.id)];
 
-	// eslint-disable-next-line @typescript-eslint/require-await
-	public async verifyTransaction(_context: TransactionVerifyContext): Promise<VerificationResult> {
+	public async verifyTransaction(context: TransactionVerifyContext): Promise<VerificationResult> {
+		const { transaction, networkIdentifier } = context;
+		const store = context.getStore(this.id, STORE_PREFIX_AUTH);
+		const senderAccount = await store.getWithSchema<AuthAccount>(
+			transaction.senderAddress,
+			authAccountSchema,
+		);
+
+		if (transaction.nonce < senderAccount.nonce) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new InvalidNonceError(
+					`Transaction with id:${transaction.id.toString('hex')} nonce is lower than account nonce`,
+					transaction.nonce,
+					senderAccount.nonce,
+				),
+			};
+		}
+		const transactionBytes = transaction.getSigningBytes();
+
+		if (
+			transaction.moduleID === this.id &&
+			this.commands.find(cmd => cmd.id === transaction.commandID)
+		) {
+			const { mandatoryKeys, optionalKeys } = codec.decode<Keys>(
+				registerMultisignatureParamsSchema,
+				transaction.params,
+			);
+
+			// For multisig registration we need all signatures to be present (including sender's one that's why we add 1 to the count)
+			const numberOfExpectedKeys = mandatoryKeys.length + optionalKeys.length + 1;
+			if (numberOfExpectedKeys !== transaction.signatures.length) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error(
+						`There are missing signatures. Expected: ${numberOfExpectedKeys} signatures but got: ${transaction.signatures.length}.`,
+					),
+				};
+			}
+
+			// Check if empty signatures are present
+			if (!transaction.signatures.every(signature => signature.length > 0)) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('A valid signature is required for each registered key.'),
+				};
+			}
+
+			try {
+				// Verify first signature is from senderPublicKey
+				validateSignature(
+					TAG_TRANSACTION,
+					networkIdentifier,
+					transaction.senderPublicKey,
+					transaction.signatures[0],
+					transactionBytes,
+					transaction.id,
+				);
+
+				// Verify each mandatory key signed in order
+				validateKeysSignatures(
+					TAG_TRANSACTION,
+					networkIdentifier,
+					mandatoryKeys,
+					transaction.signatures.slice(1, mandatoryKeys.length + 1),
+					transactionBytes,
+					transaction.id,
+				);
+
+				// Verify each optional key signed in order
+				validateKeysSignatures(
+					TAG_TRANSACTION,
+					networkIdentifier,
+					optionalKeys,
+					transaction.signatures.slice(mandatoryKeys.length + 1),
+					transactionBytes,
+					transaction.id,
+				);
+			} catch (error) {
```

Comment:
I think this is more explicit and better, but even if error is not caught, the error will be treated as FAIL outside of this function =)

Reply:
ah yeah true, actually I had pending case in mind so was trying handle and reform the return statuses for failed cases too